components/concurrency_manager/src/lib.rs

```diff
@@ -79,6 +79,7 @@ impl ConcurrencyManager {
             result[index] = MaybeUninit::new(self.lock_table.lock_key(key).await);
         }
         #[allow(clippy::unsound_collection_transmute)]
+        #[allow(clippy::transmute_undefined_repr)]
```

Comment:
It seems also to be a UB

Reply:
I don't understand. Why is it a UB to transmute from `Vec<MaybeUninit<T>>` to `Vec<T>`?

Reply:
don't understand. Not sure if it's due to the clippy's check is more strict or the compile really changes the memory layout.

Reply:
Ok I read the type of `result` wrong

Reply:
Ah, I know why it is UB. It is indeed possible that `Wrapper<MaybeUninit<T>>` may be different from `Wrapper<T>`. https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#layout has an example of `Option`.

Actually this function `lock_keys` is only used in tests and I don't believe `Vec` can have similar behavior as `Option` XD

Reply:
got it